### PR TITLE
[fix]: [PIE-4700]: New Error Code For ImportFlow of Pipelines and InputSet for New Git-Sync Exp.

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes:
New Error Code For Import Flow is created.

CodeBash Hash Checklist: 

1. Is it new task type? (If yes, no need to answer 2 then). ->NO
2. Did you add a new class which is added in kryo serialization. ->NO 
 if yes then do not use that class in any of the delegate task payload till it makes to delegate release.  NA 
3. Did you make changes to the code in a backward-compatible manner? NA 
4. will the new version of the delegate receive old version task and process? NA 
5. If new version of the manager receives response for the old task will it be able to handle it successfully? NA 
6. Ng backward compatibility → did you make a change in delegate task which will be created in ng services via grpc call to manager ->NO
7. If the older version of ng service creates the task parameter and calls newer cg manager, will it be okay NA
8. if the newer version of ng service creates the task parameter and calls older cg manager, will it be okay NA
9. Did you remove an enum → this is not backward compatible yet (tracked by  ) ->NO
10. Did you add an enum → make sure that the enum constant is added in the end of the already defined constants. If you don’t do this then you will have issues where wrong enum type will be deserialized in delegate side.   ->  **A new Error Code DUPLICATE_FILE_IMPORT is added in ErrorCode.java, thats why CodeBase Hash fails.**

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35980)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/35980)

**Comment by vikyathharekal** [comment]
trigger codebasehash

**Comment by senjucanon2** [comment]
trigger codebasehash

**Comment by senjucanon2** [comment]
trigger codebasehash

**Comment by senjucanon2** [comment]
Is it new task type? (If yes, no need to answer 2 then). ->NO

Did you add a new class which is added in kryo serialization. ->NO

if yes then do not use that class in any of the delegate task payload till it makes to delegate release.  NA

Did you make changes to the code in a backward-compatible manner? NA

will the new version of the delegate receive old version task and process? NA

If new version of the manager receives response for the old task will it be able to handle it successfully? NA

Ng backward compatibility → did you make a change in delegate task which will be created in ng services via grpc call to manager ->NO

If the older version of ng service creates the task parameter and calls newer cg manager, will it be okay NA

if the newer version of ng service creates the task parameter and calls older cg manager, will it be okay NA

Did you remove an enum → this is not backward compatible yet (tracked by  ) ->NO

Did you add an enum → make sure that the enum constant is added in the end of the already defined constants. If you don’t do this then you will have issues where wrong enum type will be deserialized in delegate side. ->NO

File: 980-commons/src/main/java/io/harness/exception/DuplicateFileImportException.java [hunk 1]
```diff
@@ -9,10 +9,13 @@
 
 import io.harness.annotations.dev.HarnessTeam;
 import io.harness.annotations.dev.OwnedBy;
+import io.harness.eraro.ErrorCode;
 
 @OwnedBy(HarnessTeam.PIPELINE)
-public class GitYamlException extends InvalidRequestException {
-  public GitYamlException(String message) {
-    super(message);
+public class DuplicateFileImportException extends InvalidRequestException {
```
Comment (namanvermah) [inline 980-commons/src/main/java/io/harness/exception/DuplicateFileImportException.java:15]: Why InvalidRequestException over here?

Comment (namanvermah) [reply 980-commons/src/main/java/io/harness/exception/DuplicateFileImportException.java:15]: The error code should be fixed for this exception, not coming from the constructor params

**Review by namanvermah — CHANGES_REQUESTED** [review]

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35980)* status: 6 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`980-commons/src/main/java/io/harness/exception/DuplicateFileImportException.java` line 15 at r3](https://reviewable.io/reviews/harness/harness-core/35980#-N8YLe9cQ0sxPKwUEmhC-r3-15:-N8bpEqY1QKjKe5VcxHk:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1eddcc4b12326cfd3c4e0674cd352ccacf2aba2b/980-commons/src/main/java/io/harness/exception/DuplicateFileImportException.java#L15)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

The error code should be fixed for this exception, not coming from the constructor params
</blockquote></details>

Done.

**Review by namanvermah — DISMISSED** [review]
> Reviewed 1 of 4 files at r2, 1 of 2 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35980)* status: 6 file(s) to review. 1 of 2 approvals obtained.

**Review by archit-harness — DISMISSED** [review]
> Reviewed 1 of 4 files at r2, 1 of 2 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35980)* status: :shipit: complete! ✓

**Comment by mohitg0795** [comment]
*[`980-commons/src/main/java/io/harness/exception/DuplicateFileImportException.java` line 18 at r4](https://reviewable.io/reviews/harness/harness-core/35980#-N8cZBjL7BrVb1XJuVmo:-N8cZBjL7BrVb1XJuVmp:b-q5gksq) ([raw file](https://github.com/harness/harness-core/blob/14af221211a2c2a53b2e9b2491488b8dfe5e3523/980-commons/src/main/java/io/harness/exception/DuplicateFileImportException.java#L18)):*
> ```Java
> @OwnedBy(HarnessTeam.PIPELINE)
> public class DuplicateFileImportException extends WingsException {
>   private static final String MESSAGE_ARG = "message";
> ```

no spacing?

**Review by mohitg0795 — COMMENTED** [review]
> Reviewed 1 of 4 files at r2, 1 of 2 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35980)* status: :shipit: complete! ✓

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35980)* status: 1 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`980-commons/src/main/java/io/harness/exception/DuplicateFileImportException.java` line 18 at r4](https://reviewable.io/reviews/harness/harness-core/35980#-N8cZBjL7BrVb1XJuVmo:-N8cZtVc5X0MtnDjmgds:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/14af221211a2c2a53b2e9b2491488b8dfe5e3523/980-commons/src/main/java/io/harness/exception/DuplicateFileImportException.java#L18)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

no spacing?
</blockquote></details>

Done.

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35980)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Comment by senjucanon2** [comment]
trigger codebasehash

**Comment by senjucanon2** [comment]
trigger codebasehash

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35980)* status: :shipit: complete! ✓

**Review by namanvermah — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35980#-:-N8gUINU4lodXAnHkKp2:bnfp4nl)* status: :shipit: complete! ✓

**Review by mohitg0795 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35980#-:-N8gY0G91AjReGZW6ioC:bnfp4nl)* status: :shipit: complete! ✓

**Review by prashantsharma04 — APPROVED** [review]
> Reviewed 1 of 4 files at r2, 1 of 2 files at r3, 3 of 4 files at r4, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35980)* status: :shipit: complete! ✓

**Pull request merged**